src/main/kotlin/com/fasterxml/jackson/module/kotlin/ReflectionCache.kt

```diff
@@ -49,6 +52,37 @@ internal class ReflectionCache(reflectionCacheSize: Int) {
     fun kotlinFromJava(key: Method): KFunction<*>? = javaMethodToKotlin.get(key)
             ?: key.kotlinFunction?.let { javaMethodToKotlin.putIfAbsent(key, it) ?: it }
 
+    /**
+     * return null if...
+     * - can't get kotlinFunction
+     * - contains extensionReceiverParameter
+     * - instance parameter is not companion object or can't get
+     */
+    @Suppress("UNCHECKED_CAST")
+    fun valueCreatorFromJava(_withArgsCreator: AnnotatedWithParams): ValueCreator<*>? = when (_withArgsCreator) {
+        is AnnotatedConstructor -> {
+            val constructor = _withArgsCreator.annotated as Constructor<Any>
+
+            javaConstructorToValueCreator.get(constructor)
+                ?: kotlinFromJava(constructor)?.let {
+                    val value = ConstructorValueCreator(it)
+
+                    javaConstructorToValueCreator.putIfAbsent(constructor, value) ?: value
+                }
```

Comment:
Just a suggestion to remove the blank line (same in the next one). I was tempted to suggest another use of `.let {}` but I nested `let` is a bit silly.

```suggestion
            javaConstructorToValueCreator.get(constructor)
                ?: kotlinFromJava(constructor)?.let {
                    val value = ConstructorValueCreator(it)
                    javaConstructorToValueCreator.putIfAbsent(constructor, value) ?: value
                }
```